mvhashmap_test.go

```diff
@@ -2,26 +2,174 @@ package block_stm
 
 import (
 	"fmt"
-	"github.com/stretchr/testify/require"
+	"math/rand"
 	"testing"
+
+	"github.com/stretchr/testify/require"
 )
 
+// create data for a given txIdx and incarnation
 func valueFor(txIdx, inc int) []byte {
 	return []byte(fmt.Sprintf("%ver:%ver:%ver", txIdx*5, txIdx+inc, inc*5))
 }
 
+// Result of the following benchmark test:
+// goos: darwin
+// goarch: arm64
+// pkg: github.com/paulgoleary/go-block-stm
+// BenchmarkWriteTimeSameLocationDifferentTxIdx-8   	       1	1383456542 ns/op	256013944 B/op	10000068 allocs/op
+// PASS
+// ok  	github.com/paulgoleary/go-block-stm	1.793s
+func BenchmarkWriteTimeSameLocationDifferentTxIdx(b *testing.B) {
+	mvh2 := MakeMVHashMap()
+	ap2 := []byte("/foo/b")
+	b.ResetTimer()
+	for i := 0; i < 1000000; i++ {
+		txIdx := rand.Intn(1000000000000000)
+		mvh2.Write(ap2, Version{txIdx, 1}, valueFor(txIdx, 1))
+	}
+}
+
+// Result of the following benchmark test:
+// goos: darwin
+// goarch: arm64
+// pkg: github.com/paulgoleary/go-block-stm
+// BenchmarkReadTimeSameLocationDifferentTxIdx-8   	1000000000	         0.8775 ns/op	       0 B/op	       0 allocs/op
+// PASS
+// ok  	github.com/paulgoleary/go-block-stm	156.475s
+func BenchmarkReadTimeSameLocationDifferentTxIdx(b *testing.B) {
+	mvh2 := MakeMVHashMap()
+	ap2 := []byte("/foo/b")
+	txIdxSlice := []int{}
+	for i := 0; i < 1000000; i++ {
+		txIdx := rand.Intn(1000000000000000)
+		txIdxSlice = append(txIdxSlice, txIdx)
+		mvh2.Write(ap2, Version{txIdx, 1}, valueFor(txIdx, 1))
+	}
+
+	b.ResetTimer()
+	for _, value := range txIdxSlice {
+		mvh2.Read(ap2, value)
```

Comment:
This can be optimized by compiler and removed completely as useless code.

Reply:
I am sorry, didn't get this. Could you please elaborate?